Fatras/include/ActsFatras/EventData/Particle.hpp

```diff
@@ -256,6 +308,8 @@ class Particle {
   // accumulated material
   Scalar m_pathInX0 = Scalar(0);
   Scalar m_pathInL0 = Scalar(0);
+  // reference surface
+  std::shared_ptr<const Acts::Surface> m_referenceSurface;
```

Comment:
I wonder if we need to store a `std::shared_ptr` here... This should never be a temporary surface, but always a surface belonging to the tracking geometry, right?
So it should be enough to store a `const Surface *` here I think...